### ISSUE TITLE
LSP Support

### ISSUE THREAD

**Issue by JakeStanger** [JakeStanger/corn#11]
There should be an LSP uh, thing, to provide full support to VSCode, Vim and all the rest.

**Comment by yavko** [comment]
Schema validation could be added to this, cuz most serialization formats seem to

**Comment by JakeStanger** [comment]
Yes I've had schema validation in mind for a long time, but it's never gotten further than a vague thought. I'm not sure what the syntax should be yet - perhaps a "special" input would work, called `$$schema`? This would keep it consistent with JSON, and avoid collisions with any real input names.

```corn
let {
  $$schema = "https://json-schema.org/draft/2019-09/schema"
} in {
  // blah
}
```
